### PR TITLE
Add explanation for deadbeef reference.

### PR DESCRIPTION
**Pull request by stefanroelofs**
I did not know what deadbeef ([wiki](https://en.wikipedia.org/wiki/Deadbeef)) implied, so it looked funny to me.
I added some explanation for this hex value.

Plus some minor markdown linting.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -238,24 +241,31 @@ The same thing can also be expressed as a filter:
 - `area_id(lookup_value)` returns the area ID for a given device ID, entity ID, or area name. Can also be used as a filter.
 - `area_name(lookup_value)` returns the area name for a given device ID, entity ID, or area ID. Can also be used as a filter.
 
+<div class='note'>
+
+Recent versions of Home Assistant generate the area ID from the area name. So Living Room becomes `living_room`.
+Before the area IDs were automatically generated and may look like this: `deadbeefdeadbeefdeadbeefdeadbeef`.
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown:247]: We don't keep a changelog in our documentation.

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -215,6 +216,8 @@ The same thing can also be expressed as a filter:
 - `is_device_attr(device_or_entity_id, attr_name, attr_value)` returns whether the value of `attr_name` for the given device or entity ID matches `attr_value`. Not supported in [limited templates](#limited-templates).
 - `device_id(entity_id)` returns the device ID for a given entity ID or device name. Can also be used as a filter.
 
+Device IDs are automatically generated when you add a new device, and may look like this: `deadbeefdeadbeefdeadbeefdeadbeef`.
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown:219]: How is this relevant here?

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/configuration/templating.markdown (hunk 2 above)
Comment (stefanroelofs) [reply source/_docs/configuration/templating.markdown:219]: The `deadbeefdeadbeefdeadbeefdeadbeef` string is used in the examples below as an example device id. This looked very funny to me, so here's an explanation what this string means.

File: source/_docs/configuration/templating.markdown (hunk 1 above)
Comment (stefanroelofs) [reply source/_docs/configuration/templating.markdown:247]: Agreed, but this is only acknowledging that for some (most?) users the area id will be a generated identifier.
You have a suggestion?

File: source/_docs/configuration/templating.markdown (hunk 2 above)
Comment (frenck) [reply source/_docs/configuration/templating.markdown:219]: Well, to me this whole line feels out of scope and out of the blue in the context of this page.

File: source/_docs/configuration/templating.markdown (hunk 1 above)
Comment (frenck) [reply source/_docs/configuration/templating.markdown:247]: Yes, remove it, as we only reflect the latest state in our documentation.
